dev_shutdown.groovy

```diff
@@ -0,0 +1,122 @@
+            script {
+              wrap([$class: 'BuildUser']) { def user = env.BUILD_USER }
+              if (user == null) {
+                user = 'auto-start'
+              }
+              currentBuild.description = "DESTROY! | USER: ${user}"
+              env.WORKSPACE = pwd()
+              deleteDir()
+
+              if (!repoFunctionsFactory.checkoutGitRepo(
+                  gitlab.cvr_terraform.url,
+                  tf_branch,
+                  gitlab.cvr_terraform.name,
+                  globalValuesFactory.SSH_DEPLOY_GIT_CREDS_ID
+              )) {
+                failure("Failed to clone terraform repository ${gitlab.cvr_terraform.url}; branch: ${tf_branch}")
+              }
+
+              dir('terraform/etc/') {
+                all_envs_raw = findFiles(glob: 'env_*.tfvars')
+              }
+              for (env in all_envs_raw) {
+                String env_name = env.name.toString().split('_')[2].minus('.tfvars')
+                envs_list << env_name
+              }
+              String[] all_exclusions = params.exclusions_raw.split(',')
+              for (String excl in all_exclusions) {
+                exclusions << ("${excl}").toString().trim()
+              }
+              println "ALL ENVS: ${envs_list}"
+              println "EXCLUSIONS: ${exclusions}"
+              for (String env in envs_list) {
+                println "> Verifying environment: ${env}"
+                if (! exclusions.contains(env)) {
+                  println ">> Marked to be lulled"
+                  lull_list << env 
+                }
+              }
+              println "[INFO] Environments to be lulled: ${lull_list}"
+            } //script
+          } //steps
+        } //stage pick DEV envs
+
+        stage('lull DEV envs') {
+          failFast true
+          steps {
+            script {
+              Map output = awsFunctionsFactory.awsCli(
+                "aws sts get-caller-identity --query 'Account' --output text"
+              )
+              if (output.status) {
+                failure("Failure while retrieving AWS account ID")
+              }
+              account_id = output.stdout.trim()
+
+              for (String env in lull_list) {
+                projectBucketPrefix = "${project}-${account_id}-${globalValuesFactory.AWS_REGION}-${env}"
+                s3DeploymentBucket = "${projectBucketPrefix}-deployment"
+                
+                def bucket_info = awsFunctionsFactory.awsCli("aws s3 ls s3://${s3DeploymentBucket} --region ${globalValuesFactory.AWS_REGION}")
+                if (bucket_info.status == 0 ) {
+                  println "> Destroying '${env}' environment..."
+                  build job: 'CVR/CVR_build', parameters: [string(name: 'ENVIRONMENT', value: env), string(name: 'ACTION', value: 'destroy')], propagate: false
+                } else { 
+                  println "> Environment '${env}' does not exist. Skipping..."
+                }
+              }
```

Comment:
pretty long block, would be nice to comment that ends the `for` loop for those that do not use IDE